ts/sagas/wallet/pagopaApis.ts

```diff
@@ -360,6 +366,86 @@ export function* updateWalletPspRequestHandler(
   }
 }
 
+/**
+ * delete all those payment methods that have the specified function enabled
+ * @param pagoPaClient
+ * @param pmSessionManager
+ * @param action
+ */
+export function* deleteAllPaymentMethodsByFunctionRequestHandler(
+  pagoPaClient: PaymentManagerClient,
+  pmSessionManager: SessionManager<PaymentManagerToken>,
+  action: ActionType<typeof deleteAllPaymentMethodsByFunction>
+): Generator<Effect, void, any> {
+  const deleteAllByFunctionApi = pagoPaClient.deleteAllPaymentMethodsByFunction(
+    action.payload as string
+  );
+  const deleteAllByFunctionApiWithRefresh = pmSessionManager.withRefresh(
+    deleteAllByFunctionApi
+  );
+
+  try {
+    const deleteResponse: SagaCallReturnType<typeof deleteAllByFunctionApiWithRefresh> = yield call(
+      deleteAllByFunctionApiWithRefresh
+    );
+    if (deleteResponse.isRight() && deleteResponse.value.status === 200) {
+      const notDeletedMethodsCount =
+        deleteResponse.value.value.data?.notDeletedWallets ?? -1;
+      // some error occurred while deletion
+      if (notDeletedMethodsCount !== 0) {
+        yield put(
+          deleteAllPaymentMethodsByFunction.failure({
+            error: Error("can't delete some methods"),
+            notDeletedMethodsCount
+          })
+        );
+        return;
+      }
+      const deletedMethodsCount =
+        deleteResponse.value.value.data?.deletedWallets ?? -1;
+
+      // this should be avoided, see https://pagopa.atlassian.net/browse/PM-150
+      const maybeWallets: SagaCallReturnType<typeof getWallets> = yield call(
+        getWallets,
+        pagoPaClient,
+        pmSessionManager
+      );
+      if (maybeWallets.isRight()) {
+        const successAction = deleteAllPaymentMethodsByFunction.success({
+          wallets: maybeWallets.value,
+          deletedMethodsCount
+        });
+        yield put(successAction);
+      } else {
+        yield put(
+          deleteAllPaymentMethodsByFunction.failure({
+            error: maybeWallets.value,
+            notDeletedMethodsCount
```

Comment:
Why a `Left` response from `getWallets` trigger a `deleteAllPaymentMethodsByFunction.failure` action? 
Is it expected? If yes can you add a comment?

Reply:
Because this API should does 2 things:

1. delete payment methods
2. return the update list of the payment methods
As mentioned some line above this API should include both of these ops
`// this should be avoided, see https://pagopa.atlassian.net/browse/PM-150`
So at the moment app does both and if one of them fails we consider it an error

Reply:
I added a comment
d5819dadee643b7f495c4d86d3d4131d3690fa75